### PR TITLE
[mono] Implement AssemblyBuilder.GetSatelliteAssembly

### PR DESCRIPTION
**Pull request by alexischr**
Contributes to https://github.com/mono/mono/issues/14788

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs [hunk 1]
```diff
@@ -546,22 +546,43 @@ public override Module[] GetLoadedModules(bool getResourceModules)
...
         {
-            throw new NotImplementedException();
-#if FALSE
-			StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
-			return GetSatelliteAssembly (culture, version, true, ref stackMark);
-#endif
+            return InternalGetSatelliteAssembly(culture, version, true)!;
+        }
+
+        internal Assembly? InternalGetSatelliteAssembly(CultureInfo culture, Version? version, bool throwOnFileNotFound)
```
Comment (marek-safar) [inline src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs:559]: Why internal with `throwOnFileNotFound` parameter which is always true?

**Comment by lambdageek** [comment]
This version of `InternalGetSatelliteAssembly` is based on https://github.com/dotnet/runtime/blob/5cd7e970487c4cc9ae652185a2f867159c48c9a0/src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L367-L395 right?

Can we share the naming logic in one place somewhere between both runtimes?

**Comment by alexischr** [comment]
@lambdageek Yeah it looks like this particular method is very similar in CoreCLR. I think I can just move just it to a new file `src/libraries/System.Reflection/src/RuntimeAssembly.cs` and use it in both, but not so much with the rest of the file.

**Comment by alexischr** [comment]
@lambdageek Turns out sharing code here isn't possible easily after all, the CoreCLR version depends on a couple of QCalls and we don't have anything similar or worth emulating for just this bit of code.

**Comment by lambdageek** [comment]
> Turns out sharing code here isn't possible easily after all, the CoreCLR version depends on a couple of QCalls and we don't have anything similar or worth emulating for just this bit of code.

I don't understand that - all the logic around forming the assembly name is identical - it's not possible to pull that out into an `internal` helper method and call it from both places?

**Comment by alexischr** [comment]
@lambdageek Are you looking at : https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L553 and not the mono copy you linked above? It is very similar but calls to `GetPublicKey()` `GetFlags()`, `GetSimpleName()` and `GetVersion()`  are in there, and `Load` is replaced by a different `InternalLoad` qcall in CoreCLR. So it just looks awkward for me to solve this by adding compatible wrappers for these to Mono.

**Comment by alexischr** [comment]
@lambdageek Just pushed an update to at least share the Mono code

**Review by lambdageek — APPROVED** [review]

**Comment by lambdageek** [comment]
> @lambdageek Are you looking at : https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L553 and not the mono copy you linked above?

Doh! yes, I was looking at the Mono copy.

The new version where the Mono code is shared LGTM.

Sorry about the confusion

**Pull request merged**